sdk/Sdk/Abstractions/SdkFunctionMetadata.cs

```diff
@@ -23,5 +23,7 @@ internal class SdkFunctionMetadata
         public IDictionary<string, object> Properties { get; set; } = new Dictionary<string, object>();
 
         public Collection<ExpandoObject> Bindings { get; set; } = new Collection<ExpandoObject>();
+
+        public SdkRetryOptions? Retry { get; set; }
```

Comment:
We need to update the source gen version of metadata generation with this prop as well. Correct? Perhaps a follow up PR later.